docs/config_entries_config_flow_handler.md

```diff
@@ -83,6 +83,37 @@ if device_unique_id:
 await self._async_handle_discovery_without_unique_id()
 ```
 
+### Unique ID Requirements
+
+A Unique ID is used to match a config entry to the underlying device or api. The Unique ID must be stable and should not be able ot be changed by the user. The Unique ID can be used to update the config entry data when device access details change. For example, for devices that communicate over the local network, if the IP address changes due a new DHCP assigment, the integration can use the Unique ID to update the host using the following code snippet:
+
+```
+    await self.async_set_unique_id(serial_number)
+    self._abort_if_unique_id_configured({CONF_HOST: host, CONF_PORT: port})
+```
+
+#### Example acceptable sources for a unique ID:
+
+- Serial number of a device
+- MAC address: formatted using `homeassistant.helpers.device_registry.format_mac`
```

Comment:
Should we mention that we want to avoid using the ip address to look up MAC address (via `getmac`) as that doesn't work in all network environments?

Reply:
I added some wording about where its acceptable to get the mac address from.